### PR TITLE
Add extension for Intel's Task Sequence

### PR DESCRIPTION
**Pull request by tiwaria1**
This extension adds new instructions that allow calling functions asynchronously from invocations.

### CODE REVIEW

**Comment by tiwaria1** [comment]
@bashbaug please help me get this merged, thank you! :)

**Comment by Naghasan** [comment]
Hi @tiwaria1, could you add the html of your extension and the update README with a link to the html preview please. You can look at this [PR](https://github.com/KhronosGroup/SPIRV-Registry/pull/240) as an example.

**Comment by tiwaria1** [comment]
> Hi @tiwaria1, could you add the html of your extension and the update README with a link to the html preview please. You can look at this [PR](https://github.com/KhronosGroup/SPIRV-Registry/pull/240) as an example.

Yes absolutely I'll add that, sorry I missed it in this attempt.

File: extensions/INTEL/SPV_INTEL_task_sequence.asciidoc [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+
+== Contributors
+
+- Jessica Davies, Intel +
+- Joe Garvey, Intel +
+- Robert Ho, Intel +
+- Michael Kinsner, Intel +
+- Abhishek Tiwari, Intel +
+- Bowen Xue, Intel
+== Notice
```
Comment (Naghasan) [inline extensions/INTEL/SPV_INTEL_task_sequence.asciidoc:21]:
```suggestion

== Notice
```

File: extensions/INTEL/SPV_INTEL_task_sequence.asciidoc [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+2+^|*OpTaskSequenceReleaseINTEL* +
+Release the memory allocated for the task sequence uniquely identified by the
+id _Sequence_.
+_Sequence_ must have type *OpTypeTaskSequenceINTEL*.
+| Capability:
+*TaskSequenceINTEL*
+| 2 | 6166 | _<id>_ +
+_Sequence_
+|=====
+== SPIR-V Representation in LLVM IR
```
Comment (Naghasan) [inline extensions/INTEL/SPV_INTEL_task_sequence.asciidoc:465]:
```suggestion

== SPIR-V Representation in LLVM IR
```

File: extensions/INTEL/SPV_INTEL_task_sequence.asciidoc [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+This is a non-normative section. `OpTypeTaskSequenceINTEL` can be mapped to LLVM
+opaque type `spirv.TaskSequenceINTEL` and mangled as
+`\\__spirv_TaskSequenceINTEL__`.
+== Issues
```
Comment (Naghasan) [inline extensions/INTEL/SPV_INTEL_task_sequence.asciidoc:469]:
```suggestion

== Issues
```

File: extensions/INTEL/SPV_INTEL_task_sequence.asciidoc [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+None.
+== Revision History
```
Comment (Naghasan) [inline extensions/INTEL/SPV_INTEL_task_sequence.asciidoc:471]:
```suggestion

== Revision History
```

File: extensions/INTEL/SPV_INTEL_task_sequence.asciidoc [hunk 5]
```diff
@@ -0,0 +1,478 @@
...
+--
+
+=== Type Declaration Instruction
+
+Add a new subsection, 3.42.26, Task Sequence Type Declaration Instruction, and
+add one new instruction in this subsection as follows:
+
+[cols="3", width="100%"]
+|=====
+2+^|*OpTypeTaskSequenceINTEL* +
```
Comment (Naghasan) [inline extensions/INTEL/SPV_INTEL_task_sequence.asciidoc:333]:
```suggestion
2+|*OpTypeTaskSequenceINTEL* +
```

**Review by Naghasan — COMMENTED** [review]
Thanks for the additions, you have rendering issues though.

I made suggestions for the headers (just missing an empty line). For the instruction tables you added a `^` causing the text to be centered, I highlighted one but it seems they are all affected.

**Comment by tiwaria1** [comment]
> Thanks for the additions, you have rendering issues though.
> 
> I made suggestions for the headers (just missing an empty line). For the instruction tables you added a `^` causing the text to be centered, I highlighted one but it seems they are all affected.

@Naghasan Please take a look again, I have removed the centering from the Instructions, Type Declaration and the second column of the Capabilities table (left the first column centered since other extensions seem to do that too)

**Comment by Naghasan** [comment]
Nice, thanks. But it seems you didn't push the updated HTML.

**Comment by tiwaria1** [comment]
> Nice, thanks. But it seems you didn't push the updated HTML.

Done :)

**Review by Naghasan — APPROVED** [review]
Vendor extension, so not looked in detail beyond rendering

**Comment by Naghasan** [comment]
Are you planing on updating https://github.com/KhronosGroup/SPIRV-Headers/pull/323 ? Would be nice if it could be done before the next WG meeting so both goes in at the same time.

**Review by bashbaug — APPROVED** [review]
LGTM!

**Comment by bashbaug** [comment]
Merging as discussed in the March 27th teleconference.

**Pull request merged**